File: top-interview-150/0027-remove-element/solution.py
# 27. Remove Element
# https://leetcode.com/problems/remove-element/
# Accepted: 2026-09-01T17:31:39.000Z
# Language: Python3
# Collection: top-interview-150
# Runtime: 0 ms · Beats 100%
# Memory: 19.3 MB · Beats 57.58%
# Submission: https://leetcode.com/submissions/detail/2127554199/

class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        k = 0
        for num in nums : 
            if num != val : 
                nums[k] = num
                k+=1
        return k

File: top-interview-150/0027-remove-element/solution.cpp
// 27. Remove Element
// https://leetcode.com/problems/remove-element/
// Accepted: 2026-09-24T22:39:13.000Z
// Language: C++
// Collection: top-interview-150
// Runtime: 0 ms · Beats 100%
// Memory: 11.6 MB · Beats 83.91%
// Submission: https://leetcode.com/submissions/detail/2152468640/

class Solution {
public:
    int removeElement(vector<int>& nums, int val) {
        int k = 0 ;
        for (int i = 0; i<nums.size() ; i++){
            if (nums[i]!=val){
                nums[k] = nums[i] ; 
                k++;
            }
        }
        return k ;
    }
};
